src/comm.cuh

```diff
@@ -158,25 +165,24 @@ std::vector<comm_handle_t> recv_data_by_offset(
  * @returns                        Merged device buffer. The user of this function is responsible for freeing this
  *                                 returned buffer using RMM_FREE.
  */
-void* merge_free_received_offset(
-                                 std::vector<void *> received_data,
-                                 const std::vector<int64_t> &bucket_count,
-                                 size_t item_size,
-                                 int64_t &total_count,
-                                 Communicator *communicator=nullptr,
-                                 bool self_free=true)
+rmm::device_buffer
+merge_free_received_offset(
+    std::vector<void *> received_data,
```

Comment:
The latest state of the code, this is still being passed by value. 
```suggestion
    std::vector<void *> const& received_data,
```

Reply:
Sorry if I missed anything, but I am really confused. I think I am passing it by reference at the latest state of the code [here](https://github.com/rapidsai/distributed-join/blob/9ce2080062df4b8d58f2524be4c601f321328e1a/src/comm.cuh#L170)?

Reply:
I don't know how, but somehow I was reviewing a stale version of the code.